hedera-node/hapi-utils/src/main/java/com/hedera/node/app/hapi/utils/ethereum/EthTxData.java

```diff
@@ -385,7 +386,8 @@ private static EthTxData populateLegacyEthTxData(RLPItem rlpItem, byte[] rawTx)
         if (vBI.compareTo(BigInteger.valueOf(34)) > 0) {
             // after EIP155 the chain id is equal to
             // CHAIN_ID = (v - {0,1} - 35) / 2
-            chainId = vBI.subtract(BigInteger.valueOf(35)).shiftRight(1).toByteArray();
+            chainId = BigIntegers.asUnsignedByteArray(
+                    vBI.subtract(BigInteger.valueOf(35)).shiftRight(1));
```

Comment:
Consider leaving a comment to explain the reason here is that signed/unsigned high order byte thing of `BigInteger`.  It won't be obvious to someone coming across this later.  A pointer to the issue #15953 would be fine.

Reply:
@david-bakin-sl Will this description be sufficient? https://github.com/hashgraph/hedera-services/pull/15997/files#diff-ce266e5cc986c3cb7c3505dd7203a63ffacfe9df6b36cac3f80dca625380ab56R389

Reply:
Yes! Except at L389 I think you meant `The BigIntegers library` with an s ....